docs/authentication.rst

```diff
@@ -280,3 +288,130 @@ to :term:`User` identification against local accounts in `Magpie` using :envvar:
 .. todo::
 
     Support ``Authorization: Basic <base64-user-pass>`` (see `#255 <https://github.com/Ouranosinc/Magpie/issues/255>`_)
+
+
+.. _user_registration:
+
+User Registration
+------------------------
+
+.. versionadded:: 3.13
+
+.. warning::
+    Please refer to the **Security Notice** under :envvar:`MAGPIE_USER_REGISTRATION_ENABLED` regarding the impact
+    of this change.
+
+Using this feature, new :term:`User` accounts can be created by anyone (self-registration) instead of previously needed
+administrative access to create them. The new accounts will require an unique ``user_name`` and valid ``email`` in order
+to confirm and complete registration. The provided ``email`` is employed to send a notification and validation endpoint
+of the specified registration information.
+
+This operation must be enabled in the corresponding `Magpie` instance using :envvar:`MAGPIE_USER_REGISTRATION_ENABLED`
+definition in :ref:`config_user_register_approval` settings for this feature to be available. Otherwise, the only
+method to create new :term:`User` entries will remain through membership to :envvar:`MAGPIE_ADMIN_GROUP`
+(see also :ref:`perm_route_access`). Administrators can always employ this direct :term:`User` creation mean regardless
+of the enabled state of this feature.
+
+Procedure
+~~~~~~~~~~
+
+Following is the summarized notification workflow to complete registration.
+
+.. cannot use the 'term' in replace, or it breaks the reference link creation
+.. _proc_user_registration:
+.. |proc_user_registration| replace:: User Registration Procedure
+
+.. container:: bordered-caption
+
+    |proc_user_registration|
+
+.. container:: bordered-content
+    :name: proc_user_registration_block
+
+
+    1. The procedure always start at reception of a new user account as requested from the `Magpie` UI frontpage
+       (see the ``Register`` button to retrieve the submission form), or through the corresponding API request.
+       Minimal validations of submitted fields and to avoid conflicts with existing :term:`User` is accomplished
+       before moving on to the next step.
+
+    2. Upon reception of a valid new registration, an email will be sent to the submitted email such that the
+       :term:`Pending User` can validate its address with the provided confirmation link. This step is
+       always required.
+
+    3. Once the validation is received, there are two possible outcomes according to configured value of
+       :envvar:`MAGPIE_USER_REGISTRATION_APPROVAL_ENABLED`.
+
+        A) When :envvar:`MAGPIE_USER_REGISTRATION_APPROVAL_ENABLED` is **NOT** activated, only email validation is
+           required from the confirmation email. Receiving the confirmation in that case immediately completes
```

Comment:
The "Receiving" term is not clear here. User receiving the confirmation email is not enough, it must follow the confirmation link.

Reply:
see updated wording